lib/ts/Layout.cc

```diff
@@ -41,134 +41,96 @@ Layout::get()
 }
 
 void
-Layout::create(const char *prefix)
+Layout::create(ts::string_view const &prefix)
 {
   if (layout == nullptr) {
     layout = new Layout(prefix);
   }
 }
 
-static char *
-layout_relative(const char *root, const char *file)
+static void
```

Comment:
Why not return the length of the result? That would be useful to avoid `strlen` later when it's redundant.